### DISCUSSION TITLE
Prefetch does not work when button with wire:click.prefetch has a dom element surrounding the button text (Copy paste simple example)

### DISCUSSION THREAD

**Discussion by isaackearl** [livewire/livewire#3124] [7. Bugs (archived)]
### Description

This bug took me awhile to nail down... truly a strange bug.

If you have a button, and in that button the "slot" or "text" has a paragraph tag around it.. then prefetching doesn't work.  When I say it doesn't work, what I mean is that it STILL makes a request when you click instead of using the prefetched content.  The prefetching of the content itself seems to work fine, it just doesn't respect that it's already been requested.  It is hard to explain but very easy to show.  Look at the code below to see what I mean!

### Exact steps to reproduce
This example is basically copied straight out of the docs on prefetching so I'm not doing anything out of the ordinary here.  What you'll see is a hidden content with 2 buttons, one that works and one that doesn't.  The only different between the one that doesn't work and the example straight out of the documentation, is that there is a <p> tag in the button.

Copy this code:

```
class TesterLive extends Component
{

    public $contentIsVisible = false;

    public function toggleContent() {
        $this->contentIsVisible = ! $this->contentIsVisible;
    }

    public function render()
    {
        return view('livewire.tester-live');
    }
}
```
and the view:

```
<div>
    <button wire:click.prefetch="toggleContent"
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">
        Show Content
    </button>
    <button wire:click.prefetch="toggleContent"
            class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">
        <p>Show Content but prefetch doesn't work</p>{{--paragraph tag makes this not work--}}
    </button>
    @if ($contentIsVisible)
        <div class="text-lg">
            Some Content...
        </div>
    @endif
</div>
```

### Other Notes: 
This bug makes pre-fetching nearly impossible with anything that isn't a straightforward button.  If you have blade components that have if statements and other features (such as breadcrumbs etc), then any amount of nesting within the component causes prefetching to stop being respected (it pre-fetches, but then still makes a request when you click).. removing the nesting within the button or link and wa-la! it starts to work again... I've tried writing wire:key on the outmost element of both the content being fetched as well as the button... but it doesn't seem to help.

**Context**
- Livewire version: v2.2.9
- Laravel version: v8.8.0
- Browser: Chrome (probably any)

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by isaackearl** [comment]
REOPEN

**Comment by isaackearl** [comment]
this is a bug I see all over my applications.  I like to use prefetch alot for buttons that navigate pages, and unless i use a very simple button then this bug usually effects it and causes a second request to happen.  I plan on eventually trying to help fix this myself if nobody can look at the example I provided, but I am currently working 60+ hours a week and simply haven't had the time yet.

**Comment by isaackearl** [comment]
REOPEN

**Comment by mokhosh** [comment]
OMG, I was just about to start debugging the same situation, and you saved me a lot of time.
My first guess was it might be because I have `wire:loading` on the inside elements, which are svg icons btw (a very common scenario).

Anyway, any ideas how we can fix this?

**Comment by mokhosh** [comment]
Tried to PR a failing test but I failed. Added this to `tests/Browser/Prefetch/Test.php`:
```php
public function test_prefetch_where_the_button_is_wrapping_html_elements()
{
    $this->browse(function ($browser) {
        Livewire::visit($browser, Component::class)
            ->assertSeeIn('@count', '1')
            ->mouseover('@button-wraping-elements')
            ->pause(250) // We have to pause because prefetching doesn't call normal response hooks.
            ->assertSeeIn('@count', '1')
            ->click('@button-wraping-elements')
            ->assertSeeIn('@count', '2');
    });
}
```
And added a button to `tests/Browser/Prefetch/view.blade.php`:
```html
<div>
    <button wire:click.prefetch="$refresh" dusk="button">inc</button>
    <button wire:click.prefetch="$refresh" dusk="button-wraping-elements"><div>inc</div></button>

    <span dusk="count">{{ app('session')->get('count') }}</span>
</div>
```

But I'm getting an unrelated error, apparently because I don't know how to run `dusk` tests outside a Laravel project:
```
1) Tests\Browser\Prefetch\Test::test
Facebook\WebDriver\Exception\SessionNotCreatedException: session not created: Chrome version must be between 70 and 73
  (Driver info: chromedriver=2.45.615355 (d5698f682d8b2742017df6c81e0bd8e6a3063189),platform=Mac OS X 10.16.0 x86_64)
```

**Comment by jkudish** [comment]
Can confirm I'm having the exact same issue, with just an svg icon inside the `<button>`. 

Pre-fetched content not used, it does 2 requests:

```php
        <button
            wire:key="assignee-dropdown-button"
            type="button"
                class="text-left inline-flex w-full bg-white text-sm font-medium text-gray-700 focus:outline-none group"
                wire:click.prefetch="$set('expanded', {{ ! $expanded }})"
        >
                <x-heroicon-o-user-add
                    class="rounded-full border-2 border-dashed w-7 h-7 p-1 text-gray-500 group-hover:border-indigo-500 group-hover:text-indigo-500"/>
        </button>
        
         @if($expanded)
                <div>
                // ...
                </div>
        @endif
```

If I change it to simple text inside the button, it works as expected:

```php
        <button
            wire:key="assignee-dropdown-button"
            type="button"
                class="text-left inline-flex w-full bg-white text-sm font-medium text-gray-700 focus:outline-none group"
                wire:click.prefetch="$set('expanded', {{ ! $expanded }})"
        >
                Click me!
        </button>
        
         @if($expanded)
                <div>
                // ...
                </div>
        @endif
```

Any ideas @calebporzio ?
